src/file-append-promise.ts

```diff
@@ -18,25 +18,15 @@ const prepareOptions = (opts: Partial<WriteOptions>) => {
 const fileAppendPromise = (
   path: string,
   data: string,
-  options?: Partial<WriteOptions>
+  options: Partial<WriteOptions>
 ) => {
   return new Promise((resolve, reject) => {
-    const appendCallback = (path: string) => (err: Error) => {
-      if (err) {
-        return reject(err)
-      }
-      return resolve(path)
+    try {
+      fs.appendFileSync(path, data, prepareOptions(options))
```

Comment:
Are we sure this will actually solve the problem here? [`rowWriterMapper`](https://github.com/cimasim89/flat-file-creator/blob/24387040bf9462536ec1884e3f9882b49ee4c3ad/src/get-async-flat-file-creator.ts#L5) is still calling `fileAppendPromise` once for every row, which still creates an async operation for each row and leaves open the possibility of out-of-order rows, doesn't it?

It seems like if we want to guarantee the order of the file (which I think is a good idea), we should probably change this to be a serialized operation, which we might do by implementing it using a [writable stream](https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options). That would take a bit more of a refactor, but seems like a really solid solution. What do you think?